website/sales/admin.py

```diff
@@ -0,0 +1,560 @@
+        return super().changelist_view(request, extra_context)
+
+    def num_items(self, obj):
+        return obj.num_items
+
+    def discount(self, obj):
+        if obj.discount:
+            return f"€{obj.discount:.2f}"
+        return "-"
+
+    def total_amount(self, obj):
+        if obj.total_amount:
+            return f"€{obj.total_amount:.2f}"
+        return "-"
+
+    def paid(self, obj):
+        if obj.total_amount is None or obj.total_amount == 0:
+            return None
+        return obj.payment is not None
+
+    paid.boolean = True
+
+    def age_restricted(self, obj):
+        return bool(obj.age_restricted) if obj else None
+
+    age_restricted.boolean = True
+
+
+@register(Product)
+class ProductAdmin(admin.ModelAdmin):
+    list_display = ("name", "age_restricted")
+
+
+class ProductListItemInline(admin.TabularInline):
+    model = ProductListItem
+    extra = 0
+
+
+@register(ProductList)
+class ProductListAdmin(admin.ModelAdmin):
+    inlines = [
+        ProductListItemInline,
+    ]
+
+
+class OrderInline(admin.TabularInline):
+    model = Order
+    ordering = ("created_at",)
+    extra = 0
+    show_change_link = True
+    can_delete = False
+
+    fields = (
+        "created_at",
+        "id",
+        "order_description",
+        "discount",
+        "total_amount",
+        "paid",
+        "payer",
+    )
+
+    readonly_fields = (
+        "created_at",
+        "id",
+        "order_description",
+        "discount",
+        "total_amount",
+        "paid",
+        "payer",
+    )
+
+    def has_add_permission(self, request, obj):
+        return False
+
+    def has_change_permission(self, request, obj=None):
+        if obj and obj.locked:
+            return False
+
+        if obj and obj.is_manager(request.member):
+            return False
+
+        return super().has_change_permission(request, obj)
+
+    def get_queryset(self, request):
+        queryset = super().get_queryset(request)
+        queryset = queryset.select_properties(
+            "total_amount", "subtotal", "num_items", "age_restricted"
+        )
+        queryset = queryset.prefetch_related("payment")
+        queryset = queryset.prefetch_related("order_items__product__product")
+        queryset = queryset.prefetch_related("payer")
+        return queryset
+
+    def total_amount(self, obj):
+        if obj.total_amount:
+            return f"€{obj.total_amount:.2f}"
+        return "-"
+
+    def discount(self, obj):
+        if obj.discount:
+            return f"€{obj.discount:.2f}"
+        return "-"
+
+    def paid(self, obj):
+        if obj.total_amount is None or obj.total_amount == 0:
+            return None
+        return obj.payment is not None
+
+    paid.boolean = True
+
+
+@register(Shift)
+class ShiftAdmin(admin.ModelAdmin):
+    inlines = [
+        OrderInline,
+    ]
+    search_fields = (
+        "id",
+        "custom_title",
+        "start",
+        "end",
+    )
+    date_hierarchy = "start"
```

Comment:
```suggestion
    filter_horizontal = ("managers",)
    date_hierarchy = "start"
```

Reply:
Wow sick, didn't know this existed. I thought I had to define a whole admin form with widgets for this